### PR TITLE
feat(ec2): disable api termination

### PR DESCRIPTION
**Pull request by AlemanCS**
### Issue # (if applicable)

None

### Reason for this change

Cloudformation has `disableApiTermination` property for turning on Termination Protection on EC2 Instance.
This property is not configurable from L2.

### Description of changes

Add `disableApiTermination` property to InstanceProps.

### Description of how you validated changes

I've added both unit and integ tests.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk-testing/framework-integ/test/aws-ec2/test/integ.instance-disable-api-termination.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    instance.connections.allowFromAnyIpv4(ec2.Port.icmpPing());
+  }
+}
+
+const testCase = new TestStack(app, 'integ-ec2-instance-disable-api-termination');
+
+new IntegTest(app, 'instance-disable-api-termination', {
+  testCases: [testCase],
+});
```
Comment (nmussy) [inline packages/@aws-cdk-testing/framework-integ/test/aws-ec2/test/integ.instance-disable-api-termination.ts:42]: Could you add a `awsApiCall` assertion to make sure that the property is being set? See [INTEGRATION_TESTS](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md#assertions) and [DescribeInstanceAttribute](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeInstanceAttribute.html)

File: packages/aws-cdk-lib/aws-ec2/lib/instance.ts [hunk 2]
```diff
@@ -317,6 +317,19 @@ export interface InstanceProps {
    */
   readonly ebsOptimized?: boolean;
 
+  /**
+   * If you set this parameter to true , you can't terminate the instance using the Amazon EC2 console, CLI, or API;
+   *
+   * otherwise, you can. To change this attribute after launch, use [ModifyInstanceAttribute](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_ModifyInstanceAttribute.html).
+   * Alternatively, if you set InstanceInitiatedShutdownBehavior to terminate, you can terminate the instance
+   * by running the shutdown command from the instance.
```
Comment (nmussy) [inline packages/aws-cdk-lib/aws-ec2/lib/instance.ts:325]: Your wording and formatting seemed a bit off, but feel free to update my suggestion further. And just to be clear, is `instanceInitiatedShutdownBehavior` going to override this parameter? If that's the case, it should be made more explicit here.

```suggestion
   * If true, the instance will not be able to be terminated using the Amazon EC2 console, CLI, or API.
   *
   * To change this attribute after launch, use [ModifyInstanceAttribute](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_ModifyInstanceAttribute.html).
   * Alternatively, if you set InstanceInitiatedShutdownBehavior to terminate, you can terminate the instance
   * by running the shutdown command from the instance.
```

**Review by nmussy — CHANGES_REQUESTED** [review]
Thanks for the contribution. Just a couple of changes, let me know if you need help with the assertion 👍

File: packages/@aws-cdk-testing/framework-integ/test/aws-ec2/test/integ.instance-disable-api-termination.ts (hunk 1 above)
Comment (AlemanCS) [reply packages/@aws-cdk-testing/framework-integ/test/aws-ec2/test/integ.instance-disable-api-termination.ts]: Added assertion

File: packages/aws-cdk-lib/aws-ec2/lib/instance.ts (hunk 2 above)
Comment (AlemanCS) [reply packages/aws-cdk-lib/aws-ec2/lib/instance.ts]: Modified comment

**Review by mazyu36 — COMMENTED** [review]
Is it possible to re-run the integ test?
It seems the test is failing due to changes in the snapshot caused by #30108. 
The issue should be resolved by running the integ test again and updating the snapshot.​​​​​​​​​​​​​​​​


```sh
@aws-cdk-testing/framework-integ:   CHANGED    aws-ec2/test/integ.instance-disable-api-termination 1.291s
@aws-cdk-testing/framework-integ:       Mappings
@aws-cdk-testing/framework-integ: [~] Mapping LatestNodeRuntimeMap: {"af-south-1":{"value":"nodejs20.x"},"ap-east-1":{"value":"nodejs20.x"},"ap-northeast-1":{"value":"nodejs20.x"},"ap-northeast-2":{"value":"nodejs20.x"},"ap-northeast-3":{"value":"nodejs20.x"},"ap-south-1":{"value":"nodejs20.x"},"ap-south-2":{"value":"nodejs20.x"},"ap-southeast-1":{"value":"nodejs20.x"},"ap-southeast-2":{"value":"nodejs20.x"},"ap-southeast-3":{"value":"nodejs20.x"},"ap-southeast-4":{"value":"nodejs20.x"},"ca-central-1":{"value":"nodejs20.x"},"cn-north-1":{"value":"nodejs18.x"},"cn-northwest-1":{"value":"nodejs18.x"},"eu-central-1":{"value":"nodejs20.x"},"eu-central-2":{"value":"nodejs20.x"},"eu-north-1":{"value":"nodejs20.x"},"eu-south-1":{"value":"nodejs20.x"},"eu-south-2":{"value":"nodejs20.x"},"eu-west-1":{"value":"nodejs20.x"},"eu-west-2":{"value":"nodejs20.x"},"eu-west-3":{"value":"nodejs20.x"},"il-central-1":{"value":"nodejs20.x"},"me-central-1":{"value":"nodejs20.x"},"me-south-1":{"value":"nodejs20.x"},"sa-east-1":{"value":"nodejs20.x"},"us-east-1":{"value":"nodejs20.x"},"us-east-2":{"value":"nodejs20.x"},"us-gov-east-1":{"value":"nodejs18.x"},"us-gov-west-1":{"value":"nodejs18.x"},"us-iso-east-1":{"value":"nodejs18.x"},"us-iso-west-1":{"value":"nodejs18.x"},"us-isob-east-1":{"value":"nodejs18.x"},"us-west-1":{"value":"nodejs20.x"},"us-west-2":{"value":"nodejs20.x"}} to {"af-south-1":{"value":"nodejs20.x"},"ap-east-1":{"value":"nodejs20.x"},"ap-northeast-1":{"value":"nodejs20.x"},"ap-northeast-2":{"value":"nodejs20.x"},"ap-northeast-3":{"value":"nodejs20.x"},"ap-south-1":{"value":"nodejs20.x"},"ap-south-2":{"value":"nodejs20.x"},"ap-southeast-1":{"value":"nodejs20.x"},"ap-southeast-2":{"value":"nodejs20.x"},"ap-southeast-3":{"value":"nodejs20.x"},"ap-southeast-4":{"value":"nodejs20.x"},"ap-southeast-5":{"value":"nodejs20.x"},"ap-southeast-7":{"value":"nodejs20.x"},"ca-central-1":{"value":"nodejs20.x"},"ca-west-1":{"value":"nodejs20.x"},"cn-north-1":{"value":"nodejs18.x"},"cn-northwest-1":{"value":"nodejs18.x"},"eu-central-1":{"value":"nodejs20.x"},"eu-central-2":{"value":"nodejs20.x"},"eu-isoe-west-1":{"value":"nodejs18.x"},"eu-north-1":{"value":"nodejs20.x"},"eu-south-1":{"value":"nodejs20.x"},"eu-south-2":{"value":"nodejs20.x"},"eu-west-1":{"value":"nodejs20.x"},"eu-west-2":{"value":"nodejs20.x"},"eu-west-3":{"value":"nodejs20.x"},"il-central-1":{"value":"nodejs20.x"},"me-central-1":{"value":"nodejs20.x"},"me-south-1":{"value":"nodejs20.x"},"mx-central-1":{"value":"nodejs20.x"},"sa-east-1":{"value":"nodejs20.x"},"us-east-1":{"value":"nodejs20.x"},"us-east-2":{"value":"nodejs20.x"},"us-gov-east-1":{"value":"nodejs18.x"},"us-gov-west-1":{"value":"nodejs18.x"},"us-iso-east-1":{"value":"nodejs18.x"},"us-iso-west-1":{"value":"nodejs18.x"},"us-isob-east-1":{"value":"nodejs18.x"},"us-west-1":{"value":"nodejs20.x"},"us-west-2":{"value":"nodejs20.x"}}
```

**Review by mazyu36 — APPROVED** [review]
Thanks. LGTM

**Pull request merged**